modules/db/src/main/java/org/mule/module/db/internal/domain/connection/DefaultDbConnection.java

```diff
@@ -363,5 +383,85 @@ public int getNetworkTimeout() throws SQLException
     {
         return delegate.getNetworkTimeout();
     }
-
+    
+    @Override
+    protected void resolveLobs(String typeName, Object[] attributes) throws SQLException {
+        ResultSet resultSet = null;
+        try
+        {
+            resultSet = this.getMetaData().getAttributes(this.getCatalog(), null, typeName, null);
+            int index = 0;
+            while (resultSet.next())
+            {
+                String dataType = resultSet.getString(ATTR_TYPE_NAME_INDEX);
+                
+                doResolveLobIn(attributes, index, dataType);
+                
+                index++;
+            }
+        }
+        finally 
+        {
+            if (resultSet != null)
+            {
+                resultSet.close();
+            }
+        }
+    }
+    
+
+    protected void doResolveLobIn(Object[] attributes, int index, String dataTypeName) throws SQLException
+    {
+        if (dataTypeName.equals(BLOB_DB_TYPE.getName()))
+        {
+            attributes[index] = createBlob(attributes[index]);
+        }
+        else
```

Comment:
should check the data type for this too?

Reply:
I've change this so that in case the dbDataType is known from the jdbc implementation, it is used to determine if the attribute has a type that has to be resolved. Otherwise the jdbc type name is used.